client/startup/notifications/soundAndDesktopNotifications.ts

```diff
@@ -0,0 +1,105 @@
+import { Meteor } from 'meteor/meteor';
+import { Session } from 'meteor/session';
+import { Tracker } from 'meteor/tracker';
+
+import { CachedChatSubscription } from '../../../app/models/client';
+import { Notifications } from '../../../app/notifications/client';
+import { readMessage } from '../../../app/ui-utils/client';
+import { KonchatNotification } from '../../../app/ui/client';
+import { getUserPreference } from '../../../app/utils/client';
+import { IMessage } from '../../../definition/IMessage';
+import { IRoom } from '../../../definition/IRoom';
+import { ISubscription } from '../../../definition/ISubscription';
+import { fireGlobalEvent } from '../../lib/utils/fireGlobalEvent';
+import { isLayoutEmbedded } from '../../lib/utils/isLayoutEmbedded';
+
+type NotificationEvent = {
+	title: string;
+	text: string;
+	duration: number;
+	payload: {
+		_id: IMessage['_id'];
+		rid: IMessage['rid'];
+		tmid: IMessage['_id'];
+		sender: IMessage['u'];
+		type: IRoom['t'];
+		name: IRoom['name'];
+		message: {
+			msg: IMessage['msg'];
+			t: string;
+		};
+	};
+};
+
+Meteor.startup(() => {
+	Tracker.autorun(() => {
+		if (!Meteor.userId()) {
+			return;
+		}
+
+		/* The message notification is controlled by the server,
+		 * after the client receives the stream there are two parts:
+		 * 1. The user should receive the desktop notification "popup"
+		 * 2. The user should be notified through sound
+		 */
+		Notifications.onUser('notification', (notification: NotificationEvent) => {
+			const hasFocus = readMessage.isEnabled();
+			const muteFocusedConversations = getUserPreference(Meteor.userId(), 'muteFocusedConversations');
+			const { rid } = notification.payload;
+			const messageIsInOpenedRoom = hasFocus && Session.get('openedRoom') === notification.payload.rid;
+
+			fireGlobalEvent('notification', {
+				notification,
+				fromOpenedRoom: messageIsInOpenedRoom,
+				hasFocus,
+			});
+
+			/*
+			 * If the user is using the embedded layout, we don't have sidebar neither the concept of more rooms,
+			 * so if somehow the user receives a notification in a room that is not open, we don't show the notification.
+			 */
+			if (isLayoutEmbedded()) {
+				if (messageIsInOpenedRoom) {
+					KonchatNotification.showDesktop(notification);
+					KonchatNotification.newMessageSound(rid);
+				}
+				return;
+			}
+
+			/*
+			 * In the other hand, if the user is using the normal layout, we have the sidebar and multiple rooms,
+			 * then want to show the notification only if the notification is not from the current one
+			 */
+
+			!messageIsInOpenedRoom && KonchatNotification.showDesktop(notification);
```

Comment:
is this a code style we are happy with? I'm personally not 😬  why not use an `if`?
```suggestion
			if (!messageIsInOpenedRoom) {
				KonchatNotification.showDesktop(notification);
			}
```